lib/options.nix

```diff
@@ -403,5 +403,31 @@ rec {
         if cfg ? lazyLoad then lib.literalMD "`false` when lazy-loading is enabled." else true;
       example = false;
     };
+
+  mkMaybeUnpackagedPackage =
+    loc: pkgs: name: package:
+    if lib.isOption package then
+      package
+    else if package != null then
+      lib.mkPackageOption pkgs name {
+        nullable = true;
+        default = package;
+      }
+    else
+      (lib.mkOption {
+        type = types.nullOr types.package;
+        description = ''
+          Package to use for ${name}.
+          Nixpkgs does not include this package, and as such an external derivation or null must be provided.
+        '';
+        default = throw ''
+          Nixvim (${loc}): No package is known for ${name}, to resolve this either:
```

Comment:
```suggestion
          Nixvim (${lib.showOption loc}): No package is known for ${name}, to resolve this either:
```

Reply:
Oh, I see. `loc` isn't a list.

Maybe we should have a different name, e.g. `fullName`?